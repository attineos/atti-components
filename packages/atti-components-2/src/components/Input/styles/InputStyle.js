import styled from 'styled-components'

const StyledInput = styled.input`
  color: ${({ theme }) => theme.components.input.colors.text};
  background-color: ${({ theme }) => theme.components.input.colors.background};
`

export default StyledInput
