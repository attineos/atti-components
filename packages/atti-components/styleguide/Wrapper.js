import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
