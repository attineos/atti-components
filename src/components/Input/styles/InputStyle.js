import styled from 'styled-components'

const StyledInput = styled.input`
  color: ${({ theme }) => theme.components.input.input.colors.text};
  background-color: ${({ theme }) => theme.components.input.input.colors.background};
`

export default StyledInput
