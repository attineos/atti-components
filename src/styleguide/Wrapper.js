import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
