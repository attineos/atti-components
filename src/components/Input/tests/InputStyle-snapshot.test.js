/* This file was generated automatically, please do not write in it. */
import React from 'react';
import InputStyle from '../styles/InputStyle';
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer';

const mock = () => {}

it('InputStyle renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <InputStyle ></InputStyle>
      </Wrapper>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
    