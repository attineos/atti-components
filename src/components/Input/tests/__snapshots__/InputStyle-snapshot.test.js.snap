// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputStyle renders correctly 1`] = `
<input
  className="InputStyle__StyledInput-sc-3f32vx-0 dMGIKm"
/>
`;
